### PR TITLE
refactor: use ButtonV2 in `src/components/[a-o]`

### PR DESCRIPTION
**Pull request by kopancek**
### Describe what this PR does

- Refactored to use ButtonV2 in `src/components/[a-o]`
- Needed to reintroduce `iconOnly` prop to ButtonV2, because the notification button was not playing ball without defining it. Maybe there's a better way though? 🤔 

### Screenshots

Screenshot 2023-12-22 at 11 03 38
Screenshot 2023-12-22 at 11 05 51
Screenshot 2023-12-22 at 11 12 11
Screenshot 2023-12-22 at 11 12 27
Screenshot 2023-12-22 at 11 17 36
Screenshot 2023-12-22 at 11 20 21
Screenshot 2023-12-22 at 11 32 28
Screenshot 2023-12-22 at 11 35 52
Screenshot 2023-12-22 at 11 38 18
Screenshot 2023-12-22 at 11 39 05
Screenshot 2023-12-22 at 11 39 59
Screenshot 2023-12-22 at 11 40 35
Screenshot 2023-12-22 at 11 43 12
Screenshot 2023-12-22 at 11 49 29
Screenshot 2023-12-22 at 11 59 14
Screenshot 2023-12-22 at 12 12 11
Screenshot 2023-12-22 at 12 24 58
Screenshot 2023-12-22 at 13 14 48
Screenshot 2023-12-22 at 13 15 12
Screenshot 2023-12-22 at 13 53 34
Screenshot 2023-12-22 at 14 16 58
Screenshot 2023-12-22 at 14 20 58
Screenshot 2023-12-22 at 14 38 40
Screenshot 2023-12-22 at 14 38 48
Screenshot 2023-12-22 at 14 39 04
Screenshot 2023-12-22 at 14 46 20
Screenshot 2023-12-22 at 14 48 57
Screenshot 2023-12-22 at 14 54 38
Screenshot 2023-12-22 at 15 03 55
Screenshot 2023-12-22 at 15 05 44
Screenshot 2023-12-22 at 15 07 50
Screenshot 2023-12-22 at 15 14 44

### On those affected packages:
- [X] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [X] MobileL (420px)
- [ ] Tablet (656px)
- [X] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

### CODE REVIEW

File: packages/shared/src/components/filters/TagCategoryDropdown.tsx [hunk 1]
```diff
@@ -77,7 +77,7 @@ export default function TagCategoryDropdown({
       <TagCategoryDetailsContent data-testid="tagCategoryTags">
         {tagCategory.tags.map((tag) => (
           <TagButton
-            className="mr-3 mb-3"
+            className="mr-3 mb-3 tag-button"
```
Comment (rebelchris) [inline packages/shared/src/components/filters/TagCategoryDropdown.tsx:80]: We wouldn't need this right?

File: packages/shared/src/components/layout/MainLayoutHeader.tsx [hunk 2]
```diff
@@ -107,6 +107,7 @@ function MainLayoutHeader({
                     secondary={atNotificationsPage}
                   />
                 }
+                iconOnly
```
Comment (rebelchris) [inline packages/shared/src/components/layout/MainLayoutHeader.tsx:110]: I think we should avoid it.
you want to check quickly online?

File: packages/shared/src/components/cards/CollectionCard/CollectionCard.spec.tsx [hunk 3]
```diff
@@ -86,6 +86,6 @@ it('should display the `collection` pill in the header', async () => {
 it('should show options button on hover when in laptop size', async () => {
   renderComponent();
   const header = await screen.findByLabelText('Options');
-  expect(header).toHaveClass('flex');
+  expect(header).toHaveClass('inline-flex');
```
Comment (rebelchris) [inline packages/shared/src/components/cards/CollectionCard/CollectionCard.spec.tsx:89]: Why was this change needed, I don't see the relevant change

**Review by rebelchris — CHANGES_REQUESTED** [review]
Minor comments, you want to quickly sync on it?

File: packages/shared/src/components/cards/CollectionCard/CollectionCard.spec.tsx (hunk 3 above)
Comment (kopancek) [reply packages/shared/src/components/cards/CollectionCard/CollectionCard.spec.tsx:89]: ButtonV2 is using `inline-flex` instead of `flex` className.

As a result, `OptionButton` is also using `inline-flex`. Hence this test was failing. Honestly I don't understand why we test for the className there in the first place, but this was a quick fix...

Comment (rebelchris) [reply packages/shared/src/components/cards/CollectionCard/CollectionCard.spec.tsx:89]: Oh why did we change it from `flex` to `inline-flex`?

**Review by rebelchris — APPROVED** [review]
Good from my side :D
Just the linting I guess :)
